### DISCUSSION TITLE
Docker Container

### DISCUSSION THREAD

**Discussion by AutomaticHomeGamer** [NillerMedDild/Enigmatica6#1361] [Ideas]
I have a working Dockerfile and run command and was wondering if there was any interest in adding it more officially here. eula.txt, settings.cfg, and world/ are all mapped from the host filesystem to the container filesystem to there is no bypassing the eula agreement and the server is still configurable and easy to back up. 

I'm happy to pass along the Dockerfile, run command, and write up some documentation on how to use it.

**Comment by NielsPilgaard** [comment]
Yes please :)

**Reply by AutomaticHomeGamer** [reply]
Docker admin commands:
----------------
Creates a new locally hosted image(in the same folder as the Dockerfile)
[Dockerfile.txt](https://github.com/NillerMedDild/Enigmatica6/files/6018215/Dockerfile.txt)
:
> docker build -t enigmatica/e6:latest .

Pull official image:
> docker pull enigmatica/6:latest

Docker user commands:
----------------
Start a container:
>docker run -d --name="Enigmatica-6" \
>-e TZ=America/Chicago \
>-p 25565:25565 \
>-v /path/on/host/eula.txt:/Enigmatica/eula.txt \
>-v /path/on/host/settings.cfg:/Enigmatica/settings.cfg \
>-v /path/on/host/world:/Enigmatica/world \
>enigmatica/e6:latest


Check status of container:
>	docker ps -a
>	docker ps -f name=Engimatica-6

See server output:
>	docker logs --follow Enigmatica-6
>	crtl+c to stop

Connect to container's shell prompt:
>	docker exec -ti Enigmatica-6 sh
>	exit to stop

Stop container:
>	docker stop Enigmatica-6

Remove container:
>	docker rm Enigmatica-6

Update image:
>	docker pull enigmatica/e6:latest


General Docker:
----------------
Docker run command info:
>	TZ: Sets the timezone/clock of the container. Should be set to the same time as your host machine. 
>	-p: Maps a host port to port in the container in the format hostPort:containerPort
>	-v: Maps a host folder/file to a container folder/file in the format hostFile:containerFile
>	--name: Names the instance of the container. 


To update your server:
>	1. Stop your container.
>	2. Remove your container.	
>	3. Update your image with a docker pull command
>	4. Start a new container with a docker run command.


The start/update stuff only really works if there is a dedicated image on DockerHub. Without it the user would need to grab the latest Dockerfile and then build the image locally before starting or updating.
